lib/ASTGen/CMakeLists.txt

```diff
@@ -32,6 +32,13 @@ if (SWIFT_SWIFT_PARSER)
     Sources/ASTGen/Types.swift
   )
 
+  # The compat56 library is not available during a stage-0 compiler build.
+  # Once we have proper stage tracking, we should turn this on for stages that
+  # are guaranteed to have the compatibility libraries.
+  target_compile_options(swiftASTGen PRIVATE
```

Comment:
Just a question here: why does this matter if the stage 0 build for ASTGen builds against the host tools? Shouldn’t the host driver provide whatever compat libs we shipped in the host’s toolchain?

It’s totally fine to set this since, yeah, we aren’t exactly trying to use bleeding-edge features in ASTGen.

Reply:
The issue is that the compatibility library isn't in the host toolchain yet (have to be able to go back two , so we don't link it because it doesn't exist, and we end up failing to link due to the missing `__swift_FORCE_LOAD_$_swiftCompatibility56` symbol;

```
__swift_FORCE_LOAD_$_swiftCompatibility56", referenced from:
      __swift_FORCE_LOAD_$_swiftCompatibility56_$_swiftASTGen in ASTGen.swift.o
      __swift_FORCE_LOAD_$_swiftCompatibility56_$_swiftASTGen in Decls.swift.o
      __swift_FORCE_LOAD_$_swiftCompatibility56_$_swiftASTGen in Diagnostics.swift.o
      __swift_FORCE_LOAD_$_swiftCompatibility56_$_swiftASTGen in Exprs.swift.o
      __swift_FORCE_LOAD_$_swiftCompatibility56_$_swiftASTGen in Generics.swift.o
      __swift_FORCE_LOAD_$_swiftCompatibility56_$_swiftASTGen in Literals.swift.o
      __swift_FORCE_LOAD_$_swiftCompatibility56_$_swiftASTGen in Macros.swift.o
```

Reply:
Wait a sec, I see your point. Looking back at the logs, it's the bootstrapping1 compiler that's failing to link. That makes more sense. The stage-0 compiler knows about this library and adds the symbols. Then the stage-0 compiler tries linking the stage-1 compiler against the hostlibs instead of the matching libraries, which don't have the compat56 library, and so that fails.